services/commonService.js

```diff
@@ -0,0 +1,82 @@
+fluid.registerNamespace("myL3");
+
+fluid.defaults("myL3.commonService", {
+    gradeNames: "fluid.component",
+    invokers: {
+        getData: {
+            funcName: "myL3.commonService.getData",
+            args: ["{arguments}.0"]
+        }
+    }
+});
+
+myL3.commonService.getData = (fields) => {
```

Comment:
`myL3.commonService.getData` is called twice:

- Once from `myL3.locationService.importLocation`
    - `const fakeArray = [];`
    - `myL3.commonService.getData(fakeArray);`
- Once from `app.js`
    - `myL3.commonService.getData(fieldsSelectedByLearner);`

Let's refactor to clean this up. It's not clear what the intend of `myL3.commonService.getData([])` is. Make separate functions with clearer names.

Reply:
Created fakeArray because we need location and we no longer have that. Now, getData() function contains the code to first import location and then weather data. Also we need weather data while importing location because that will give us the name of city and country using latitude and longitude i.e. to import location data we are bound to call both google maps (for map and marker) API and openweathermap API(for city and country name).

Reply:
I found another way in ES6. We can do something like this to avoid fakeArray dilemma:
```function getData(fluid = [])  {
    // .........
   }
```